### ISSUE TITLE
Results expectation

### ISSUE THREAD

**Issue by hannayurchyk** [DanHrmti/ECRL#2]
Hello again, 

I have a brief question about the results that I am supposed to get on wandb as well as from running `main.py` and `eval_policy.py`. 

I'm attempting to train an ordered push policy. I run `main.py -c config/ordered_push`. In the respective `Config.yaml` file I specify `latentRepPath: 'latent_rep_chkpts/dlp_push_5C'` for pre-trained DLP and train with `numObjects = 2`. I run `policy_eval.py config/ordered_push` with the respective checkpoint found in `model_chkpts` and the same number of objects. 

I am unsure of what I am supposed to expect in the results. My issues are as follows:

1. There are several `.gif` files that I obtained in wanbd: Eval Goal Image, Goal Image - Frontview, Goal Image - Sideview, Episode Video - Frontview, Episode Video - Sideview, and Eval Episode Video. I'm getting videos for both `main.py` and `policy_eval.py`. I'm not sure about what these videos represent and at which point in the training they are recorded.

2. I notice that the code runs after 800 episodes. I'm probably misunderstanding the code a little bit, but I'm currently expecting that the code should stop as it is specified in the config file. 

3. I would be curious to see what are my expected graphs for `obs_rms_std` and `obs_rms_mean` should look like. I see the error decrease, but only from 0.68 to 0.655 and -0.03 to -0.04 respectively (which tells me that I'm probably not learning much). Would you mind telling me what results you got on that end?

Thank you very much for your help! 

Hanna 

p.s. Would it be possible to email you for further personal issues I have?

**Comment by DanHrmti** [comment]
Hi Hanna,

1. Training through `main.py`: The Eval Episode Video and Goal Image are recorded every `config['Training']['epochEpisodes']` when evaluating the policy during training. The other videos and goals are recorded every `config['WANDB']['episodeVisFreq'] `. The current config, if you haven't modified it, is `800` episodes for both. Notice that the Eval video displays a policy that seems less stochastic, this is because there is no exploration noise added when evaluating.

Evaluation with `policy_eval.py`: This script evaluates a chosen policy and prints the performance statistics as well as saves a single rollout video + goal in order to visualize the policy behavior.

2. The code will run `config['Training']['totalTimesteps']` timesteps in the environment.

3. The inputs to the policy and Q-function neural networks are normalized using running statistics (running mean and std). The `obs_rms_std` and `obs_rms_mean` graphs record a mean of these across the object feature dimension and it is mostly meant to identify anomalies such as a sudden very large change in statistics that might indicate a bug. What you should expect in these graphs is relatively quick convergence to some value, 'obs_rms_std' shouldn't be too close to zero. The graphs that indicate performance are `eval_goal_achievement_%`, `mean_success_frac`, `mean_avg_obj_dist`, `mean_max_obj_dist` and `eval_mean_reward`. I would also look at `1C_interaction_rate` which can signal when the agent starts to learn that it needs to interact with objects in order to achieve its goals.

Let me know if you have any other questions. You can also email me if you prefer, I will be happy to help.

Best,
Dan
